service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureExtractor.java

```diff
@@ -84,16 +85,16 @@ private static String buildReactionDescription(AllergyIntolerance.AllergyIntoler
         return StringUtils.EMPTY;
     }
 
-    private static String buildReactionSeverity(AllergyIntolerance.AllergyIntoleranceReactionComponent reactionComponent) {
-        if (reactionComponent.hasSeverity()) {
-            return REACTION_SEVERITY + reactionComponent.getSeverity();
+    private static String buildReactionExposureRoute(AllergyIntolerance.AllergyIntoleranceReactionComponent reactionComponent) {
+        if (reactionComponent.hasExposureRoute() && extractTextOrCoding(reactionComponent.getExposureRoute()).isPresent()) {
```

Comment:
How about this?
```
return Optional.ofNullable(reactionComponent.getExposureRoute())
    .flatMap(AllergyStructureExtractor::extractTextOrCoding)
    .map(value -> REACTION_EXPOSURE_ROUTE + value)
    .orElse(StringUtils.EMPTY);
```

Reply:
because you are doing the same `extractTextOrCoding` computation twice

Reply:
From what I noticed the problem with HAPI's model implementation is that it can throw errors if `hasXXX` methods are ommited, so I'm not sure it's a wise idea to call `get` method like this.